test/fluentd-forward.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+# This is a test suite for the fluentd secure_forward feature
+
+source "$(dirname "${BASH_SOURCE[0]}" )/../hack/lib/init.sh"
+source "${OS_O_A_L_DIR}/deployer/scripts/util.sh"
+os::util::environment::use_sudo
+
+os::test::junit::declare_suite_start "test/fluentd-forward"
+
+if [ -n "${DEBUG:-}" ] ; then
+    set -x
+fi
+
+update_current_fluentd() {
+    # this will update it so the current fluentd does not send logs to an ES host
+    # but instead forwards to the forwarding fluentd
+
+    # undeploy fluentd
+    os::log::debug "$( oc label node --all logging-infra-fluentd- )"
+    os::cmd::try_until_failure "oc get pod $fpod"
+
+    # edit so we don't send to ES
+    oc get configmap/logging-fluentd -o yaml | sed '/## matches/ a\
+      <match **>\
+        @include configs.d/user/secure-forward.conf\
+      </match>' | oc replace -f -
+
+    POD=$( oc get pods -l component=forward-fluentd -o name )
+    FLUENTD_FORWARD=$( oc get $POD --template='{{.status.podIP}}' )
+
+    # update configmap secure-forward.conf
+    oc patch configmap/logging-fluentd --type=json --patch '[{ "op": "replace", "path": "/data/secure-forward.conf", "value": "\
+  @type secure_forward\n\
+  self_hostname forwarding-${HOSTNAME}\n\
+  shared_key aggregated_logging_ci_testing\n\
+  secure no\n\
+  buffer_queue_limit \"#{ENV['"'BUFFER_QUEUE_LIMIT'"']}\"\n\
+  buffer_chunk_limit \"#{ENV['"'BUFFER_SIZE_LIMIT'"']}\"\n\
+  <server>\n\
+   host '${FLUENTD_FORWARD}'\n\
+   port 24284\n\
+  </server>"}]'
+
+    # redeploy fluentd
+    os::log::debug "$( oc label node --all logging-infra-fluentd=true )"
+    os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
+    fpod=$( get_running_pod fluentd )
+}
+
+create_forwarding_fluentd() {
+    # create forwarding configmap named "logging-forward-fluentd"
+    oc create configmap logging-forward-fluentd \
+       --from-file=fluent.conf=$OS_O_A_L_DIR/hack/templates/forward-fluent.conf
+
+    # create forwarding daemonset
+    oc get daemonset/logging-fluentd -o yaml | \
+        sed -e 's/logging-infra-fluentd: "true"/logging-infra-forward-fluentd: "true"/' \
```

Comment:
Unfortunately, the sed s/ fluentd/ forward-fluentd/ changes at least a dozen lines . . . perhaps you can tell me how to use oc patch to edit this from the env section of the first container?
```
        - name: FLUENTD_CPU_LIMIT
          valueFrom:
            resourceFieldRef:
              containerName: fluentd-elasticsearch
              divisor: "0"
              resource: limits.cpu
```
I need to change the "containerName" to "forward-fluentd-elasticsearch"

Otherwise, imo, not worth it.